### ISSUE TITLE
Fails to install on windows.

### ISSUE THREAD

**Issue by mohd-taba** [originell/smaz-py3#3]
The installation via pip fails on windows; it fails to install both the wheel and via setup.py. It returns the following error.
` ValueError: path '/home/runner/work/smaz-py3/smaz-py3/smazmodule.c' cannot be absolute`
pip is updated to the latest version (pip 21.2.4).

**Comment by originell** [comment]
Thanks for reporting. That's true, I've never tested this under Windows.

Please, could you provide me with the full traceback?

I'll try to see what it takes to make it run under windows.

**Comment by originell** [comment]
Hey @mohd-taba 
I just [released a new version of smaz-py3 (1.1.0)](https://pypi.org/project/smaz-py3/1.1.0/). Can you try again if it works now?

**Comment by mohd-taba** [comment]
> 
> 
> Hey @mohd-taba
> I just [released a new version of smaz-py3 (1.1.0)](https://pypi.org/project/smaz-py3/1.1.0/). Can you try again if it works now?

Yes. It does work.
Importing and basic usage seems to have no issues as well.
Thanks.

**Comment by originell** [comment]
@mohd-taba glad to hear! happy coding!

**Comment by ExcaliburZero** [comment]
I've run into the same error trying to install smaz-py3 version 1.1.2 on Windows.

I was also able to reproduce the bug in a GitHub actions build:

https://github.com/ExcaliburZero/cbpickaxe/actions/runs/6608486205/job/17947254519

* **Python version:** 3.11
* **smaz-py3 version:** 1.1.2

```
Collecting smaz-py3 (from cbpickaxe==0.1.0)
  Downloading smaz-py3-1.1.2.tar.gz (20 kB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  Getting requirements to build wheel did not run successfully.
  exit code: 1
  
  [69 lines of output]
  C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\dist.py:498: SetuptoolsDeprecationWarning: Invalid dash-separated options
  !!
  
          ********************************************************************************
          Usage of dash-separated 'description-file' will not be supported in future
          versions. Please use the underscore name 'description_file' instead.
  
          This deprecation is overdue, please update your project and remove deprecated
          calls to avoid build errors in the future.
  
          See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
          ********************************************************************************
  
  !!
    opt = self.warn_dash_deprecation(opt, section)
  running egg_info
  writing smaz_py3.egg-info\PKG-INFO
  writing dependency_links to smaz_py3.egg-info\dependency_links.txt
  writing top-level names to smaz_py3.egg-info\top_level.txt
  reading manifest file 'smaz_py3.egg-info\SOURCES.txt'
  Traceback (most recent call last):
    File "C:\hostedtoolcache\windows\Python\3.11.6\x64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 353, in <module>
      main()
    File "C:\hostedtoolcache\windows\Python\3.11.6\x64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 335, in main
      json_out['return_val'] = hook(**hook_input['kwargs'])
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\hostedtoolcache\windows\Python\3.11.6\x64\Lib\site-packages\pip\_vendor\pyproject_hooks\_in_process\_in_process.py", line 118, in get_requires_for_build_wheel
      return hook(config_settings)
             ^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\build_meta.py", line 355, in get_requires_for_build_wheel
      return self._get_build_requires(config_settings, requirements=['wheel'])
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\build_meta.py", line 325, in _get_build_requires
      self.run_setup()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\build_meta.py", line 507, in run_setup
      super(_BuildMetaLegacyBackend, self).run_setup(setup_script=setup_script)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\build_meta.py", line 341, in run_setup
      exec(code, locals())
    File "<string>", line 14, in <module>
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\__init__.py", line 103, in setup
      return distutils.core.setup(**attrs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 185, in setup
      return run_commands(dist)
             ^^^^^^^^^^^^^^^^^^
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\_distutils\core.py", line 201, in run_commands
      dist.run_commands()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 969, in run_commands
      self.run_command(cmd)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\dist.py", line 989, in run_command
      super().run_command(command)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\_distutils\dist.py", line 988, in run_command
      cmd_obj.run()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 318, in run
      self.find_sources()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 326, in find_sources
      mm.run()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 548, in run
      self.add_defaults()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 593, in add_defaults
      self.read_manifest()
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\sdist.py", line 214, in read_manifest
      self.filelist.append(line)
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\command\egg_info.py", line 485, in append
      path = convert_path(item)
             ^^^^^^^^^^^^^^^^^^
    File "C:\Users\runneradmin\AppData\Local\Temp\pip-build-env-4zyc32la\overlay\Lib\site-packages\setuptools\_distutils\util.py", line 139, in convert_path
      raise ValueError("path '%s' cannot be absolute" % pathname)
  ValueError: path '/home/runner/work/smaz-py3/smaz-py3/smazmodule.c' cannot be absolute
  [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

Getting requirements to build wheel did not run successfully.
exit code: 1

See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
Error: Process completed with exit code 1.
```

**Comment by ExcaliburZero** [comment]
I tried installing it again on Windows from source rather than using an install via pip with pypi, and it worked without issue. (CI builds are also still failing)

**Comment by originell** [comment]
thank you for reporting. that must be a build failure on my side here then. I'll look into this. Happy though building from source worked for you.

**Comment by originell** [comment]
@ExcaliburZero I've just released v1.1.3 on pypi. Could you try if that fixes it for you?

**Comment by ExcaliburZero** [comment]
> @ExcaliburZero I've just released v1.1.3 on pypi. Could you try if that fixes it for you?

Yep I tried installing v1.1.3 using pip both on Command Prompt and also on a Windows GitHub Actions build and both worked.

Thanks!

**Comment by originell** [comment]
Happy to help!
